.github/dependabot.yml

```diff
@@ -5,11 +5,31 @@ updates:
   schedule:
     interval: weekly
   open-pull-requests-limit: 10
-  allow:
-  - dependency-type: direct
-  - dependency-type: indirect
+  groups:
+    build-and-release-dependencies:
+      # Python dependencies known to be critical to our build/release security
+      patterns:
+        - "build"
+        - "hatchling"
+    test-and-lint-dependencies:
+      # Python dependencies that are only pinned to ensure test reproducibility
+      patterns:
+        - "bandit"
```

Comment:
Is this the best way to do this? I haven't seen this feature before but can we do it by updated path perhaps? That way we don't have to keep this list updated if the corresponding requirements.txt files change...

Reply:
it's entirely possible (maybe by configuring multiple `package-ecosystem: pip`s, each one pointing to a different subdir with different requirements files) but I figured that it would look so strange (and dependabot is so vaguely documented) that I couldn't be bothered.